### PR TITLE
fix/perl-suid into devel

### PR DESCRIPTION
**Pull request by obilodeau**
going into devel

See issue [#1438: perl 5.12+ and/or modern distro and setuid script](http://packetfence.org/bugs/view.php?id=1438).

Not ready yet, build system needs to be modified so that it'll handle new subpackages more cleanly. Mimic what has been done under debian.

TODO
- ~~(rpm: bin/pfcmd.pl is still setuid)~~
- ~~(security: we should probably enforce taint-mode in the bin/pfcmd.pl #! line (perl -T I think). taint mode was forced to On when running a setuid binary before but since our little wrapper, I'm pretty it'll need to be explicit (maybe not?))~~

### CODE REVIEW

**Comment by fdurand** [comment]
get_all_perl_binaries() has been updated to remove /usr/local/pf/bin/pfcmd
Removed setuid on pfcmd.pl
Added -T to pfcmd.pl

**Comment by obilodeau** [comment]
Another thing I just thought.. the perl-suid subpackage should probably depend on packetfence >= 3.4.0

This way you cannot remove it by accident. After all, it is useless completely useless without packetfence (as opposed to docs or tests).

**Comment by obilodeau** [comment]
This branch should be re-based on stable as it currently can't merge in integration easily. Due to all the packages changes in our catalyst work..

**Comment by obilodeau** [comment]
re-based on devel. will go in integration and will be reviewed shortly.

File: addons/packages/packetfence.spec [hunk 1]
```diff
@@ -36,6 +36,7 @@ URL: http://www.packetfence.org
 AutoReqProv: 0
 BuildArch: noarch
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{rev}-root
+%define debug_package %{nil}
```
Comment (obilodeau) [inline addons/packages/packetfence.spec]: what is this for?

Comment (fdurand) [inline addons/packages/packetfence.spec]: It´s for disabling the debug package

**Comment by obilodeau** [comment]
sent in integration

**Comment by obilodeau** [comment]
alright, tested on my Fedora box with Perl 5.14.2 and good to go. I'm merging this in devel right now, we will fix the nightly build scripts next week.

**Comment by obilodeau** [comment]
another issue: generated pfcmd-suid sub-package on my fedora was noarch instead of x86-64: `packetfence-pfcmd-suid-3.6.0-0.20120903.fc17.noarch.rpm`

**Comment by obilodeau** [comment]
Looks like we are having a real problem on RHEL regarding arch-specific subpackage:

```
> Unfortuantely noarch package cannot have arch subpackage, so nimbus-theme-gnome
> must be arch as well.
```

from: https://bugzilla.redhat.com/show_bug.cgi?id=488174#c6

**Comment by obilodeau** [comment]
Ok, so it seems the recommended way to do it is a virtual top-level package with one noarch and one arch-specific subpackage. Here's [an example](https://bugzilla.redhat.com/attachment.cgi?id=344264).

**Comment by obilodeau** [comment]
With the latest changes, here's a result of the build:

```

Wrote: /home/olivier/rpmbuild/SRPMS/packetfence-3.6.0-0.20120904fc17.src.rpm
Wrote: /home/olivier/rpmbuild/RPMS/noarch/packetfence-core-3.6.0-0.20120904fc17.noarch.rpm
Wrote: /home/olivier/rpmbuild/RPMS/noarch/packetfence-remote-snort-sensor-3.6.0-0.20120904fc17.noarch.rpm
Wrote: /home/olivier/rpmbuild/RPMS/x86_64/packetfence-pfcmd-suid-3.6.0-0.20120904fc17.x86_64.rpm
```

Looking good. However I'm scared of hard-to-predict side effects of such a change.. @fdurand once you approve, let's have that in devel as soon as we can so we can adapt the build sytem and ask the community to test it out.

**Comment by obilodeau** [comment]
Jean had a good point, I wonder if someone upgrading packetfence will pull in packetfence-core...

Relevant link: http://fedoraproject.org/wiki/Packaging:Guidelines#Renaming.2FReplacing_Existing_Packages

**Comment by obilodeau** [comment]
After some tests in VMs, I decided to try renaming the virtual package (packetfence to packetfence-virtual) and rename packetfence-core to packetfence instead. This was done here: d7f95fec2ae0a9e56d2552fb9e9f1aae27878984

**Comment by obilodeau** [comment]
I just successfully upgraded a CentOS 6 VM from a packetfence stable (only packetfence package) to a splitted packetfence + packetfence-pfcmd-suid package. Setuid C wrapper is properly installed and worked.

Ready for review and should be merged in devel pretty soon. Build scripts will be adapted after.

**Pull request merged**